src/blockchain/src/structures/chain.rs

```
@@ -44,10 +47,83 @@ impl Chain {
     pub fn last_block(&self) -> Option<Block> {
         self.blocks().last().cloned()
     }
+
+    pub fn get_block_position(&self, ordinal: Ordinal) -> Result<usize, BlockchainError> {
+        if ordinal >= self.len() as Ordinal {
+            return Err(BlockchainError::InvalidBlockchainLength(self.len()));
+        }
+
+        if ordinal > self.last_block().unwrap().header.ordinal {
+            return Err(BlockchainError::InvalidBlockchainLength(self.len()));
+        }
+
+        if ordinal >= usize::MAX as Ordinal - 1 {
+            return Err(BlockchainError::InvalidBlockchainOrdinal(ordinal));
+        }
+
+        if self.blocks[ordinal as usize].header.ordinal != ordinal {
+            return Err(BlockchainError::InvalidBlockchainOrdinal(ordinal));
+        }
+
+        Ok(ordinal as usize)
+    }
+
+    pub async fn save_block(
+        &self,
+        start: Ordinal,
+        mut end: Ordinal,
+        file_path: String,
+    ) -> Result<(), BlockchainError> {
+        match self.last_block() {
+            None => return Err(BlockchainError::InvalidBlockchainLength(self.len())),
+            Some(block) => {
+                if start > block.header.ordinal {
+                    return Err(BlockchainError::InvalidBlockchainLength(self.len()));
+                }
+
+                if end > block.header.ordinal {
+                    warn!("The end ordinal {:?} out of bounds ", end);
```

Why not just return an error in this case as well?